zipkin-storage/elasticsearch/src/test/java/zipkin2/elasticsearch/integration/ITEnsureIndexTemplate.java

```diff
@@ -0,0 +1,126 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.elasticsearch.integration;
+
+import com.linecorp.armeria.common.AggregatedHttpRequest;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.RequestHeaders;
+import java.io.IOException;
+import java.util.List;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.TestInstance;
+import zipkin2.CheckResult;
+import zipkin2.Span;
+import zipkin2.elasticsearch.ElasticsearchStorage;
+import zipkin2.elasticsearch.internal.Internal;
+
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+import static zipkin2.TestObjects.LOTS_OF_SPANS;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+abstract class ITEnsureIndexTemplate {
+
+  ElasticsearchStorage storage;
+
+  /**
+   * Returns a new {@link ElasticsearchStorage.Builder} for connecting to the backend for the test.
+   */
+  protected abstract ElasticsearchStorage.Builder newStorageBuilder(TestInfo testInfo);
+
+  @BeforeAll void setUpCatchAllTemplate(TestInfo testInfo) throws IOException {
+    ElasticsearchStorage.Builder builder = newStorageBuilder(testInfo);
+    storage = builder.build();
+    assumeTrue(storage.version() >= 7.8f, () -> "Applicable only for ES >= 7.8, skipping test");
+
+    /**
+     * Delete all index templates in order to create the "catch-all" index template, because
+     * ES does not allow multiple index templates of the same index_patterns and priority
+     */
+    deleteIndexTemplate("*");
+    setUpCatchAllTemplate();
+  }
+
+  @Test void createZipkinIndexTemplate_getTraces_returnsSuccess() throws IOException {
+    /** Create index template with {@link ElasticsearchStorage#ensureIndexTemplates()} */
+    CheckResult check = storage.check();
+
+    assertThat(check.ok()).isTrue();
+
+    List<String> traceIds = asList(LOTS_OF_SPANS[0].traceId(), LOTS_OF_SPANS[1].traceId());
+    List<Span> spans = asList(LOTS_OF_SPANS[0], LOTS_OF_SPANS[1]);
+
+    storage.spanConsumer().accept(spans).execute();
+
+    assertThat(storage.traces().getTraces(traceIds).execute())
```

Comment:
verify the catch-all template works? you might want to make something more interesting so that you can use ES HTTP query on it. ex this one https://gist.github.com/adriancole/1af1259102e7a2da1b3c9103565165d7